src/main/pages/base_page.py

```diff
@@ -82,17 +82,31 @@ def switch_to_iframe(self, by_locator):
     def return_from_iframe(self):
         self.driver.switch_to.default_content()
 
-    def switch_to_new_window(self):
+    def wait_for_element_to_disappear(self, by_locator, timeout=10):
+        WebDriverWait(self.driver, timeout).until(
+            expected_conditions.invisibility_of_element_located(by_locator)
+        )
+
+    def switch_to_new_window(self, by_locator):
         main_window = self.driver.current_window_handle
+
         all_windows = self.driver.window_handles
         for window in all_windows:
             if window != main_window:
                 self.driver.switch_to.window(window)
+                self.wait_for_element_to_disappear(by_locator)
                 break
 
-    def switch_to_tab(self, tab_index):
+    def switch_to_tab(self, tab_index, by_locator):
```

Comment:
Rename the function to express the intention to switch to a tab and wait for an element to disappear. Another thing is renaming the variable to understand that it is the locator for the element you want to wait to disappear.

Reply:
I considered to express the intention in the HomePage file where is called this function. Besides, I decides to call the function that waits until the preloader disappears out of the switch function and put it in the HomePage within the function that makes these two actions, switch and wait.